src/psi/j/executors/flux.py

```diff
@@ -0,0 +1,278 @@
+"""This module contains the Flux :class:`~psi.j.JobExecutor`."""
+
+from __future__ import annotations
+
+import time
+import logging
+import copy
+import json
+from functools import partial
+
+from typing import Any, Optional, Dict
+
+from distutils.version import StrictVersion
+
+from psi.j import InvalidJobException
+from psi.j import Job, JobExecutorConfig, JobState, JobStatus, JobAttributes
+from psi.j import JobExecutor, ResourceSpecV1
+
+logger = logging.getLogger(__name__)
+
+
+class FluxJobExecutor(JobExecutor):
+    """
+    A job executor that runs jobs via Flux.
+    """
+
+    import radical.utils as _ru
+    import flux as _flux
+
+    _NAME_ = 'flux'
+    _VERSION_ = StrictVersion('0.0.1')
+
+    # map flux states to jpsi states.
+    _state_map = {'NEW': JobState.QUEUED,
+                  'DEPEND': None,
+                  'SCHED': None,
```

Comment:
I think QUEUED in PSI/J means that the job was sent to the backend. From that perspective, the original version seems more appropriate.

Reply:
Agree with @hategan: jpsi considers the job `QUEUED` as soon as it sent off to Flux.  From the spec:
```
When the job is accepted by the backend for execution, it will enter the state QUEUED.
```